model/pod/podutil.go

```diff
@@ -0,0 +1,105 @@
+package pod
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/pkg/errors"
+)
+
+// Constants representing default curl retry arguments.
+const (
+	curlDefaultNumRetries = 10
+	curlDefaultMaxSecs    = 100
+)
+
+// CurlCommand returns the command to curl the evergreen client and retries the request with the default retry parameters.
+func (p *Pod) CurlCommand(settings *evergreen.Settings) ([]string, error) {
+	var retryArgs string
+	if curlDefaultNumRetries != 0 && curlDefaultMaxSecs != 0 {
+		retryArgs = " " + curlRetryArgs(curlDefaultNumRetries, curlDefaultMaxSecs)
+	}
+	cmds, err := p.curlCommands(settings, retryArgs)
+	if err != nil {
+		return []string{}, errors.WithStack(err)
+	}
+	return []string{"CMD-SHELL", strings.Join(cmds, " && ")}, nil
+}
+
+// AgentCommand returns the arguments to start the agent.
+func (p *Pod) AgentCommand(settings *evergreen.Settings) []string {
+	return []string{
+		fmt.Sprintf("./%s", p.BinaryName()),
+		"agent",
+		fmt.Sprintf("--api_server=%s", settings.ApiUrl),
+		fmt.Sprintf("--mode=pod"),
+		fmt.Sprintf("--pod_id=%s", p.ID),
+		fmt.Sprintf("--pod_secret=%s", p.Secret),
+		fmt.Sprintf("--log_prefix=%s", filepath.Join(p.TaskContainerCreationOpts.WorkingDir, "agent")),
+		fmt.Sprintf("--working_directory=%s", p.TaskContainerCreationOpts.WorkingDir),
+		"--cleanup",
+	}
+}
+
+// ClientURL returns the URL used to get the latest Evergreen client version
+// directly from the Evergreen server.
+func (p *Pod) ClientURL(settings *evergreen.Settings) string {
+	return strings.Join([]string{
+		strings.TrimSuffix(settings.ApiUrl, "/"),
+		strings.TrimSuffix(settings.ClientBinariesDir, "/"),
+		p.ExecutableSubPath(),
+	}, "/")
+}
+
+// S3ClientURL returns the URL in S3 where the Evergreen client version can be
+// retrieved for this server's particular Evergreen build version.
+func (p *Pod) S3ClientURL(settings *evergreen.Settings) string {
+	return strings.Join([]string{
+		strings.TrimSuffix(settings.PodInit.S3BaseURL, "/"),
+		evergreen.BuildRevision,
+		p.ExecutableSubPath(),
+	}, "/")
+}
+
+// BinaryName returns the file name of the binary.
+func (p *Pod) BinaryName() string {
+	name := "evergreen"
+	if p.TaskContainerCreationOpts.OS == OSWindows {
+		return name + ".exe"
+	}
+	return name
+}
+
+// ExecutableSubPath returns the directory containing the compiled agents.
+func (p *Pod) ExecutableSubPath() string {
```

Comment:
`clientURLSubpath` - the significance of this field is slightly different than on EC2 hosts, but it's referring to a portion of the URL path that tells it where the agent is located.

For example, if you're downloading the agent for `GOOS="linux"` and `GOARCH="amd64"`, the path to the compiled binary is:
```go
linux_amd64/evergreen
```
And for `GOOS="windows"` `GOARCH="amd64"`, the path would be:
```go
windows_amd64/evergreen.exe
```